packages/connect/src/implementation.ts

```diff
@@ -95,13 +100,15 @@ export interface HandlerContext {
  */
 export function createHandlerContext(
   spec: { service: ServiceType; method: MethodInfo },
+  signal: AbortSignal,
```

Comment:
I am afraid I made this a bit more difficult with https://github.com/bufbuild/connect-es/pull/591 🙈 

Yes, it would be nice if we just had a single AbortSignal, but you need an Abort_Controller_ to trigger an AbortSignal, and I think `universalRequestFromNodeRequest()` should neither accept nor return an AbortController. That means we cannot avoid linking AbortSignals.

But it's not too bad: https://github.com/bufbuild/connect-es/pull/623 added `createLinkedAbortController()` for this purpose. We can merge the signal from the deadline / shutdown with the one from the request before passing it to the HandlerContext.

Reply:
If I understand correctly it is necessary to wait until #623 is merged to implement this change? :open_mouth:

Reply:
Yes, https://github.com/bufbuild/connect-es/pull/623 should be merged soon 🙂